[Bug]: DeepSeek-V4-Flash decode startup crashes with aclnnScatterNdUpdateV2 aicore exception under FULL_DECODE_ONLY cudagraph mode

## Your current environment

<details>
<summary>Environment Info</summary>

- **vllm version**: 0.21.0
- **vllm-ascend commit**: 0a79510a0 (`github/main`)
- **PyTorch**: 2.10.0
- **torch_npu**: 2.10.0
- **CANN**: 9.0.0
- **NPU Driver**: 25.2.1
- **Hardware**: Ascend910 (ascend910_9392) × 16, single node
- **OS**: Linux 5.10
- **Python**: 3.12.13

</details>

## 🐛 Describe the bug

DeepSeek-V4-Flash (w8a8-mtp) decode service fails to start when `cudagraph_mode` is set to `FULL_DECODE_ONLY`. The same configuration with `--enforce-eager` starts successfully.

### Startup command

```bash
vllm serve <model_path>     --host 0.0.0.0 --port 8200     --trust-remote-code     --max-model-len 1048576     --max-num-batched-tokens 128     --max-num-seqs 60     --gpu-memory-utilization 0.88     --tensor-parallel-size 1     --data-parallel-size 16     --enable-expert-parallel     --no-disable-hybrid-kv-cache-manager     --no-enable-prefix-caching     --quantization ascend     --compilation-config '{"cudagraph_mode": "FULL_DECODE_ONLY"}'     --speculative-config '{"num_speculative_tokens":2,"method":"mtp","enforce_eager":true}'     --additional_config '{
        "ascend_compilation_config": {"enable_npugraph_ex": true, "enable_static_kernel": false},
        "enable_cpu_binding": true,
        "multistream_overlap_shared_expert": true,
        "multistream_dsa_preprocess": false,
        "enable_flashcomm1": false,
        "enable_fused_mc2": 0,
        "enable_reduce_sample": false
    }'     --kv-transfer-config '{
        "kv_connector": "MooncakeHybridConnector",
        "kv_role": "kv_consumer",
        "kv_port": "30200",
        "engine_id": "decode",
        "kv_connector_extra_config": {
            "prefill": {"dp_size": 4, "tp_size": 4, "pp_size": 1},
            "decode": {"dp_size": 16, "tp_size": 1}
        }
    }'
```

### Error

All 16 DP workers crash during startup at `profile_cudagraph_memory()` → `_warmup_and_capture()` phase.

<details>
<summary>Full worker traceback (Worker_DP6)</summary>

```
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070] WorkerProc hit an exception.
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070] Traceback (most recent call last):
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070]   File ".../vllm_ascend/worker/model_runner_v1.py", line 4923, in _replace_gpu_model_runner_function_wrapper
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070]     yield
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070]   File ".../vllm_ascend/worker/model_runner_v1.py", line 4777, in profile_cudagraph_memory
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070]     result = GPUModelRunner.profile_cudagraph_memory(self)
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070]   File ".../vllm/v1/worker/gpu_model_runner.py", line 6092, in profile_cudagraph_memory
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070]     self._warmup_and_capture(
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070]   File ".../vllm/v1/worker/gpu_model_runner.py", line 6273, in _warmup_and_capture
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070]     self._dummy_run(
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070]   File ".../vllm_ascend/worker/model_runner_v1.py", line 3520, in _dummy_run
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070]     outputs = self._model_forward(
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070]   File ".../vllm_ascend/worker/model_runner_v1.py", line 2761, in _model_forward
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070]     hidden_states = run_model()
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070]   File ".../vllm_ascend/compilation/acl_graph.py", line 206, in __call__
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070]     with torch.npu.graph(aclgraph, pool=self.graph_pool):
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070]   File ".../torch_npu/npu/graphs.py", line 519, in __enter__
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070]     torch.npu.synchronize()
(Worker_DP6_EP6 pid=70770) ERROR 06-11 06:14:46 [multiproc_executor.py:1070] RuntimeError: The Inner error is reported as above. The process exits for this inner error, and the current working operator name is aclnnScatterNdUpdateV2.
```

</details>

### CANN debug log

<details>
<summary>Hardware-level error from /root/ascend/log/debug/plog/</summary>

```
# Root cause: DDR address out of range in MTE instruction
[ERROR] RUNTIME PrintCoreInfo: The error from device(chipId:2, dieId:0), serial number is 212,
  there is an exception of fftsplus aivector error, core id is 13, error code = 0x800000
[ERROR] RUNTIME PrintCoreInfo: The extend info: errcode:(0x800000, 0, 0)
  errorStr: The DDR address of the MTE instruction is out of range.
  fixp_error0 info: 0x600002d, fixp_error1 info: 0xd9, fsmId:0, tslot:5, thread:0, ctxid:0, blk:3, sublk:0, subErrType:4.
[ERROR] RUNTIME CheckSmmuFault: Get isSmmuFault is 1.

# The failing kernel is a Compressor SuperKernel containing ScatterNdUpdateV2
[ERROR] RUNTIME PrintErrorInfoForDavinciTask: [DFX_INFO] Aicore kernel execute failed,
  device_id=4, stream_id=12, task_id=269, fault kernel_name=Compressor_5d0f45757e9591d6e00196bb7912baa5_12449

# Operator-level error
[ERROR] RUNTIME: rtsLaunchKernelWithHostArgs execution failed, reason=aicore exception
[ERROR] ASCENDCL: aclrtLaunchKernelWithHostArgs failed, runtime result = 507015.
[ERROR] OP: OpName:[aclnnScatterNdUpdateV2_6050_ScatterNdUpdateV2AiCore] aclrtLaunchKernelWithHostArgs failed, return: 507015
[ERROR] OP: KernelLaunch failed: .../kernel/ascend910_93/scatter_nd_update_v2/ScatterNdUpdateV2_0e675ed9c31b6f8eb4fec80650b6492b.o
[ERROR] OP: Kernel Run failed. opType: 59, ScatterNdUpdateV2
[ERROR] OP: launch failed for ScatterNdUpdateV2, errno:361001.

# IDEDD exception dump files
[ERROR] IDEDD: [Dump][Exception] dump host kernel to file: .../data-dump/0/Compressor_5d0f45757e9591d6e00196bb7912baa5_12449_host.o
[ERROR] IDEDD: [Dump][Exception] dump kernel json to file: .../data-dump/0/Compressor_5d0f45757e9591d6e00196bb7912baa5.json
[ERROR] IDEDD: [Dump][Exception] dump kernel to file: .../data-dump/0/Compressor_5d0f45757e9591d6e00196bb7912baa5.o
```

</details>

### Key observations

1. **`--enforce-eager` works fine** — the issue only manifests with `cudagraph_mode: FULL_DECODE_ONLY`
2. **All 16 DP workers fail** at the same point (not a single-card issue)
3. The failing custom kernel is **`ScatterNdUpdateV2`** (from `vllm_ascend/_cann_ops_custom/vendors/custom_transformer/`)
4. The crash happens during **cudagraph memory profiling** (startup phase), not during normal inference
5. The hardware error is **"DDR address of the MTE instruction is out of range"** with **SMMU fault**, indicating an illegal memory access
6. The ScatterNdUpdateV2 op is fused into a **`Compressor` SuperKernel** at the time of the aicore exception

### Workaround

Use `--enforce-eager` to skip cudagraph capture entirely.

--- a/tests/ut/worker/a2/test_worker_multi_instance.py
+++ b/tests/ut/worker/a2/test_worker_multi_instance.py
@@ -15,6 +15,7 @@
 # limitations under the License.
 #
 
+from types import SimpleNamespace
 from unittest.mock import MagicMock, patch
 
 from vllm.config import CUDAGraphMode
@@ -56,7 +57,10 @@ class TestDetermineAvailableMemoryMultiInstance(TestBase):
 
         mock_cache_config = MagicMock()
         mock_cache_config.kv_cache_memory_bytes = None
+        mock_cache_config.gpu_memory_utilization = requested_memory / init_total_memory
         worker.cache_config = mock_cache_config
+
+        worker.model_config = SimpleNamespace(hf_config=SimpleNamespace(model_type="qwen3"))
 
         mock_snapshot = MagicMock()
         mock_snapshot.free_memory = init_free_memory
@@ -134,6 +138,67 @@ class TestDetermineAvailableMemoryMultiInstance(TestBase):
         expected = requested_memory - non_kv_cache
         self.assertEqual(result, expected)
         self.assertGreater(result, 0)
+
+    @patch("vllm_ascend.worker.worker.logger")
+    def test_deepseek_v4_compressed_skips_npugraph_memory_profile(self, mock_logger):
+        """DSV4 DSA must not run the pre-KV graph memory profiling path."""
+        total = int(64 * GiB_bytes)
+        requested_memory = int(total * 0.9)
+        init_free = int(60 * GiB_bytes)
+        non_kv_cache = int(1 * GiB_bytes)
+
+        worker = self._make_worker(requested_memory, init_free, total)
+        worker.vllm_config.compilation_config.cudagraph_mode = CUDAGraphMode.FULL_DECODE_ONLY
+        worker.model_config.hf_config.model_type = "deepseek_v4"
+        worker.model_runner.use_compress = True
+        worker.model_runner.profile_cudagraph_memory.return_value = int(2 * GiB_bytes)
+        profile_result = self._make_profile_result(
+            free_memory_after=init_free - non_kv_cache,
+            non_kv_cache_memory=non_kv_cache,
+        )
+
+        with self._patch_memory_profiling(profile_result):
+            result = worker.determine_available_memory()
+
+        worker.model_runner.profile_run.assert_called_once()
+        worker.model_runner.profile_cudagraph_memory.assert_not_called()
+        self.assertEqual(
+            worker.vllm_config.compilation_config.cudagraph_mode,
+            CUDAGraphMode.FULL_DECODE_ONLY,
+        )
+        self.assertEqual(worker.npugraph_memory_estimate, 0)
+        self.assertEqual(result, requested_memory - non_kv_cache)
+
+    @patch("vllm_ascend.worker.worker.logger")
+    def test_non_deepseek_compressed_still_profiles_npugraph_memory(self, mock_logger):
+        """The DSV4 guard must not disable graph memory profiling globally."""
+        total = int(64 * GiB_bytes)
+        requested_memory = int(total * 0.9)
+        init_free = int(60 * GiB_bytes)
+        non_kv_cache = int(1 * GiB_bytes)
+        npugraph_memory = int(2 * GiB_bytes)
+
+        worker = self._make_worker(requested_memory, init_free, total)
+        worker.vllm_config.compilation_config.cudagraph_mode = CUDAGraphMode.FULL_DECODE_ONLY
+        worker.model_runner.use_compress = True
+        worker.model_runner.profile_cudagraph_memory.return_value = npugraph_memory
+        profile_result = self._make_profile_result(
+            free_memory_after=init_free - non_kv_cache,
+            non_kv_cache_memory=non_kv_cache,
+        )
+
+        with (
+            self._patch_memory_profiling(profile_result),
+            patch(
+                "vllm_ascend.worker.worker.envs_vllm.VLLM_MEMORY_PROFILER_ESTIMATE_CUDAGRAPHS",
+                False,
+            ),
+        ):
+            result = worker.determine_available_memory()
+
+        worker.model_runner.profile_cudagraph_memory.assert_called_once_with()
+        self.assertEqual(worker.npugraph_memory_estimate, npugraph_memory)
+        self.assertEqual(result, requested_memory - non_kv_cache)
 
     @patch("vllm_ascend.worker.worker.logger")
     def test_second_instance_on_same_card_positive_kv_cache(self, mock_logger):

--- a/vllm_ascend/worker/worker.py
+++ b/vllm_ascend/worker/worker.py
@@ -375,7 +375,18 @@ class NPUWorker(WorkerBase):
             profile_torch_peak = torch.npu.memory_stats(self.device).get("allocated_bytes.all.peak", 0)
 
             npugraph_memory_estimate = 0
-            if self.vllm_config.compilation_config.cudagraph_mode != CUDAGraphMode.NONE:
+            should_profile_npugraph_memory = self.vllm_config.compilation_config.cudagraph_mode != CUDAGraphMode.NONE
+            if should_profile_npugraph_memory and getattr(self.model_runner, "use_compress", False):
+                hf_config = self.model_config.hf_config
+                if getattr(hf_config, "model_type", None) == "deepseek_v4":
+                    logger.warning_once(
+                        "Skipping ACL graph memory profiling for DeepSeek-V4 "
+                        "DSA compressed attention. Graph mode remains enabled; "
+                        "the normal ACL graph capture still runs after KV cache "
+                        "allocation."
+                    )
+                    should_profile_npugraph_memory = False
+            if should_profile_npugraph_memory:
                 npugraph_memory_estimate = self.model_runner.profile_cudagraph_memory()
 
         # Override torch_peak_increase with the pre-graph-capture value to

